docs/setup/keadm.md

```diff
@@ -96,70 +96,229 @@ KubeEdge edgecore is running, For logs visit:  /var/log/kubeedge/edgecore.log
+    | metrics-server-kubeedge                               |                 latest                   | aef0fa7a834c | 28 seconds ago | 57.2MB |
+
+
+    Make sure you change the tag of image by using its IMAGE ID to be compactable with image name in yaml file.
+
+    ```bash
+    docker tag a24f71249d69 metrics-server-kubeedge:latest
+    ```
 
 3. Apply the deployment yaml. For specific deployment documents, you can refer to https://github.com/kubernetes-sigs/metrics-server/tree/master/manifests.
 
+    Before you deploy metrics-server, you have to make sure that you deploy it on the node which has apiserver deployed on. In this case, that is the master node. As a consequence, it is needed to make master node schedulable by the following command:
+
+    ``` shell
+    kubectl taint nodes --all node-role.kubernetes.io/master-
+    ```
+   
+    Then, in the deployment.yaml file, it must be specified that metrics-server is deployed on master node.
+    (The hostname is chosen as the marked label.)
+    In **metrics-server-deployment.yaml**
+    ``` yaml
+        spec:
+          affinity:
+            nodeAffinity:
+              requiredDuringSchedulingIgnoredDuringExecution:
+                nodeSelectorTerms:
+                - matchExpressions:
+                  #Specify which label in [kubectl get nodes --show-labels] you want to match
+                  - key: kubernetes.io/hostname
+                    operator: In
+                    values:
+                    #Specify the value in key
+                    - charlie-latest
+    ```
+
 **IMPORTANT NOTE:**
 1. Metrics-server needs to use hostnetwork network mode.
 
-2. Use the image compiled by yourself and set imagePullPolicy to IfNotPresent.
+2. Use the image compiled by yourself and set imagePullPolicy to Never.
 
 3. Enable the feature of --kubelet-use-node-status-port for Metrics-server
 
+    Those settings need to be written in deployment yaml (metrics-server-deployment.yaml) file like this:
+
+    ``` yaml
+          volumes:
+          # mount in tmp so we can safely use from-scratch images and/or read-only containers
+          - name: tmp-dir
+            emptyDir: {}
+          hostNetwork: true                          #Add this line to enable hostnetwork mode
+          containers:
+          - name: metrics-server
+            image: metrics-server-kubeedge:latest    #Make sure that the REPOSITORY and TAG are correct
+            # Modified args to include --kubelet-insecure-tls for Docker Desktop (don't use this flag with a real k8s cluster!!)
+            imagePullPolicy: Never                   #Make sure that the delpoyment uses the image you built up
+            args:
+              - --cert-dir=/tmp
+              - --secure-port=4443
+              - --v=2
+              - --kubelet-insecure-tls
+              - --kubelet-preferred-address-types=InternalDNS,InternalIP,ExternalIP,Hostname
```

Comment:
r u sure we should use InternalDNS here?

Reply:
I am not sure, but actually, I used this deployment file and successfully make it work.